File: ui/src/routes/api/inbox.focus.summary.ts
import { defineApi } from '@/server/api-route'
import { json } from '@/server/http'
import { requireUser } from '@/server/api-guard'
import { focusSummary } from '@/server/inbox-focus'
// doc: The one-screen summary of the caller's focus state — what's queued,
// doc: what's in flight, what needs a decision.


export const Route = defineApi('/api/inbox/focus/summary', {
  GET: async ({ request }) => {
    const user = await requireUser(request)
    if (user instanceof Response) return user
    return json(await focusSummary(user))
  },
})

File: api/crates/talaria-routes-comms/src/inbox/inbox_focus_summary.rs
// /api/inbox/focus/summary. GET → the one-screen summary of the caller's
// focus state: how many items are queued (snoozed ones excluded), as
// {count}.

use axum::Json;
use axum::extract::State;
use axum::http::{HeaderMap, StatusCode};
use axum::response::{IntoResponse, Response};
use serde_json::json;
use talaria_error::internal;
use talaria_inbox_focus::focus_summary;
use talaria_session::require_user;
use talaria_state::AppState;

pub async fn get(State(state): State<AppState>, headers: HeaderMap) -> Result<Response, Response> {
    let user = require_user(&state, &headers).await?;
    Ok(match focus_summary(&state.pg, &user).await {
        Ok(count) => (StatusCode::OK, Json(json!({ "count": count }))).into_response(),
        Err(e) => internal("[inbox-focus] summary read failed", e),
    })
}
